prow/plugins/config.go

```diff
@@ -1791,6 +1792,23 @@ func (b *Bugzilla) OptionsForRepo(org, repo string) map[string]BugzillaBranchOpt
 	return options
 }
 
+// BranchCleaner contains the configuration for the branchcleaner plugin.
+type BranchCleaner struct {
+	// ProtectedBranches is a set of repo branches, branches in this white list
+	// are exempt from branch gc, even if the branches are already merged into the target branch
+	ProtectedBranches []string `json:"protected_branches,omitempty"`
```

Comment:
I wonder if we could come up with a different name? "Branch Protection" is a GH term (https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches), there is even a Prow component that manages it, and it would be nice to avoid confusion if possible.

Reply:
OK, so how about "AllowedBranches" ?

Reply:
Could work, although it's a little generic. Maybe `PreserveBranches`?